mw/xml_dump/iteration/iterator.py

```diff
@@ -140,6 +143,9 @@ def load_site_info(cls, element):
         namespaces = {}
 
         for sub_element in element:
+
+            if sub_element.tag == 'siteinfo':
+                return cls.load_site_info(sub_element)
```

Comment:
I'm confused about this line because it looks like you're expecting to find something like this:

```
<siteinfo>
  <siteinfo> ... </siteinfo>
</siteinfo>
```